src/databricks/labs/blueprint/wheels.py

```diff
@@ -220,17 +220,26 @@ def __init__(self, installation: Installation, product_info: ProductInfo, *, ver
         self._product_info = product_info
         self._verbose = verbose
 
-    def upload_to_dbfs(self) -> str:
+    def upload_to_dbfs(self, force_dependencies: bool = False) -> str:
         """Uploads the wheel to DBFS location of installation and returns the remote path."""
-        with self._local_wheel.open("rb") as f:
-            return self._installation.upload_dbfs(f"wheels/{self._local_wheel.name}", f)
-
-    def upload_to_wsfs(self) -> str:
+        for wheel in self._local_wheel:
+            if force_dependencies or self._product_info.product_name() in wheel.name:
+                with wheel.open("rb") as f:
+                    remote_wheel = self._installation.upload_dbfs(f"wheels/{wheel.name}", f)
+                    if self._product_info.product_name() in wheel.name:
+                        main_wheel_remote_path = remote_wheel
+        return main_wheel_remote_path
+
+    def upload_to_wsfs(self, force_dependencies: bool = False) -> str:
         """Uploads the wheel to WSFS location of installation and returns the remote path."""
-        with self._local_wheel.open("rb") as f:
-            remote_wheel = self._installation.upload(f"wheels/{self._local_wheel.name}", f.read())
-            self._installation.save(Version(self._product_info.version(), remote_wheel, self._now_iso()))
-            return remote_wheel
+        for wheel in self._local_wheel:
+            if force_dependencies or self._product_info.product_name() in wheel.name:
+                with wheel.open("rb") as f:
+                    remote_wheel = self._installation.upload(f"wheels/{wheel.name}", f.read())
```

Comment:
```suggestion
                remote_wheel = self._installation.upload(f"wheels/{wheel.name}", wheel.read_bytes())
```
btw, as long as we require a byte array, this way is shorter